services/deduplicate/index.ts

```diff
@@ -118,7 +121,7 @@ export function mainDedupe() {
   }
 
   // Loop through all query blocks and set backfilled posts in the open slots.
-  queryBlocks.forEach((queryBlock) => {
+  queryBlocks.forEach(async (queryBlock) => {
```

Comment:
While this code does look to work, eventually we will want to refactor the usage of an async callback here. Some recommendations call for use of a `for...of` loop but it can largely depend on what other functionality we have going on here.